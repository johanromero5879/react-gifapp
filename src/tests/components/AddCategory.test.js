import { shallow } from 'enzyme'

import AddCategory from '../../components/AddCategory'

describe('Testing on <AddCategory />', () => {
    const setCategories = jest.fn()

    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> )

    beforeEach( () => {
        jest.clearAllMocks() //clear all simulations 
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> )
    } )

    test('should show correclty', () => {
        expect( wrapper ).toMatchSnapshot()    
    })
    
    test('should change text input', () => {
        const input = wrapper.find('input')
        const value = 'cute anime'

        input.simulate('change', { target: { value } })

        const p = wrapper.find('p')

        expect( p.text().trim() ).toBe( value )
    })
    
    test('should not post info with submit', () => {
        const form = wrapper.find('form')

        form.simulate('submit', { preventDefault(){} })

        expect( setCategories ).not.toHaveBeenCalled()
    })

    test('should call setCategories and clear input box', () => {
        const input = wrapper.find('input')
        let value = 'love anime'

        // Simulate inputChange
        input.simulate('change', { target: { value } })

        // Simulate submit
        const form = wrapper.find('form')
        form.simulate('submit', { preventDefault(){} })

        // Check if setCategories was called
        expect( setCategories ).toHaveBeenCalled()
        expect( setCategories ).toHaveBeenCalledTimes(1)
        expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) )


        // Chek if input is empty
        expect( input.prop('value') ).toBe( '' )
    })
})