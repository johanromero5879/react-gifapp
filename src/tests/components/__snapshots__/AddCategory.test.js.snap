// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <AddCategory /> should show correclty 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
